File: 1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getSize(ListNode* head){
        if(head == nullptr){
            return 0;
        }

        int size = 0;

        while(head != nullptr){
            size++;
            head = head->next;
        }

        return size;
    }

    int getDecimalValue(ListNode* head) {
        if(head == nullptr){
            return 0;
        }
        ListNode* crr = head;
        int binaryNumber = 0;
        int power = getSize(head) - 1;
        while(crr != nullptr){
            if(crr->val == 1){
                binaryNumber += pow(2,power);
            }

            power--;
            crr = crr->next;
        }
        return binaryNumber;
    }
};

File: Linked-List/1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        if(head == null){
            return 0;
        }

        ListNode crr = head;
        int size = 0;
        int result = 0;

        while(crr != null){
            size++;
            crr = crr.next;
        }

        crr = head;

        while(crr != null){
            result += crr.val * Math.pow(2, size - 1);
            crr = crr.next;
            size--;
        }

        return result;
    }
}